File: LeetCode/cpp/198.cpp
class Solution {
public:
    int rob(vector<int> &nums) {
        int size = nums.size();
        if (size == 0) return 0;
        int first = -1, second = -1;
        for (int i = 0; i < size; ++i) {
            if (i % 2 == 0) {
                if (first == -1) first = nums[i];
                else first = max(first + nums[i], second);
            } else {
                if (second == -1) second = max(first, nums[i]);
                else second = max(second + nums[i], first);
            }
        }
        return size % 2 == 0 ? second : first;
    }
};

File: LeetCode/python3/198.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0:
            return 0
        elif n == 1:
            return nums[0]
        elif n == 2:
            return max(nums[0], nums[1])
        prev = nums[0]
        curr = max(nums[0], nums[1])
        for i in range(2, n):
            temp = curr
            curr = max(curr, prev + nums[i])
            prev = temp
        return curr
